components/server/ee/src/workspace/workspace-factory.ts

```diff
@@ -56,7 +56,29 @@ export class WorkspaceFactoryEE extends WorkspaceFactory {
                 }
             }
 
-            let ws = await this.createForCommit({span}, user, commitContext, normalizedContextURL);
+            let ws;
+            let parentPrebuild;
+
+            // Find an older full prebuild to start an incremental prebuild
+            for (const parent of (commitContext.history || [])) {
```

Comment:
Thought: May want to ask {GitHub,GitLab,Bitbucket} API for commit history here, instead of in the context parser.